src/main/java/net/citizensnpcs/npc/CitizensNPC.java

```diff
@@ -185,10 +180,9 @@ public boolean spawn(Location at) {
         at = at.clone();
         getTrait(CurrentLocation.class).setLocation(at);
 
-        entityController.spawn(at, this);
+        boolean couldSpawn = entityController.spawn(at, this);
 
         net.minecraft.server.v1_8_R3.Entity mcEntity = ((CraftEntity) getEntity()).getHandle();
-        boolean couldSpawn = !Util.isLoaded(at) ? false : mcEntity.world.addEntity(mcEntity, SpawnReason.CUSTOM);
```

Comment:
to clarify, I made that part a protected method so I could override it in the human controller.